pruntime/enclave/src/lib.rs

```diff
@@ -1185,33 +1188,28 @@ fn query(q: types::SignedQuery) -> Result<Value, Value> {
     Ok(res_value)
 }
 
-fn handle_query_receipt(opaque_query: types::OpaqueQuery, accid_origin: chain::AccountId) -> Value {
-	let res = match types::deopaque_query(opaque_query.clone()) {
-		Ok(q) => {
-			let res = match q.request {
-				Request::QueryReceipt{account, tx_hash} => {
-					//println!("request is query receipt, tx_hash={:}", tx_hash);
-					//println!("account={:?}, accid_origin = {:?}", account, AccountIdWrapper(accid_origin.clone()));
-					if account ==  AccountIdWrapper(accid_origin) {
-						let gr = GLOBAL_RECEIPT.lock().unwrap();
-						let receipts: Vec<TransactionReceipt> = gr.tx_receipts.get(&account).unwrap_or(&Vec::new()).to_vec()
-							.into_iter().filter(|x| x.contract_id == opaque_query.contract_id && x.tx_hash == tx_hash).collect::<Vec<TransactionReceipt>>();
-						serde_json::to_value(Response::QueryReceipt { receipts })
-					} else {
-						serde_json::to_value(Response::Error(Error::NotAuthorized))
-					}
-				},
-			};
-
-			res.unwrap()
-		},
-		Err(_e) => {
-			//println!("request normal query");
-			Value::Null
-		},
+fn handle_query_receipt(accid_origin: Option<chain::AccountId>, req: Request) -> Response {
+	let inner = || -> Result<Response, Error> {
+		match req {
+			Request::QueryReceipt{tx_hash} => {
+				let gr = GLOBAL_RECEIPT.lock().unwrap();
+				match gr.get_receipt(tx_hash) {
+					Some(receipt) => {
+						if receipt.account == AccountIdWrapper(accid_origin.unwrap()) {
```

Comment:
Need to check if `accid_origin` is None before `unwrap` it. This could happen if the requester didn't sign the query. We can return something like `Error::NotAuthorized` when there's no origin.